src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace System.Diagnostics.Metrics
+{
+    /// <summary>
+    /// Base class of all Metrics Instrument classes
+    /// </summary>
+#if ALLOW_PARTIALLY_TRUSTED_CALLERS
+        [System.Security.SecuritySafeCriticalAttribute]
+#endif
+    public abstract class Instrument
+    {
+#if NO_ARRAY_EMPTY_SUPPORT
+        internal static KeyValuePair<string, object?>[] EmptyTags { get; } = new KeyValuePair<string, object?>[0];
```

Comment:
readonly?

Reply:
We chatted offline we cannot use readonly on the properties.

Reply:
Yeah, I'm not sure what I was commenting on, but it doesn't make sense here.